src/components/Clock.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+type State = {
+  date: Date,
+  timerId: NodeJS.Timer | undefined,
+};
+
+export class Clock extends React.Component<{}, State> {
+  state: State = {
+    date: new Date(),
+    timerId: undefined,
+  };
+
+  componentDidMount() {
+    this.setState({
+      timerId: setInterval(() => {
+        this.setState({ date: new Date() });
+        // eslint-disable-next-line
+        console.log(this.state.date.toLocaleTimeString());
+      }, 1000),
+    });
```

Comment:
hard to read such code

```suggestion
      const timerId: setInterval(() => {
        this.setState({ date: new Date() });
        // eslint-disable-next-line
        console.log(this.state.date.toLocaleTimeString());
      }, 1000),
      
      this.setState({ timerId });
```